### PR TITLE
[Disk Manager] delete UseLightCheckpoint param from replicate task request

### PR DESCRIPTION
**Pull request by BarkovBG**

### CODE REVIEW

File: cloud/disk_manager/internal/pkg/clients/nbs/client.go [hunk 1]
```diff
@@ -77,6 +77,37 @@ func getStorageMediaKind(
 	}
 }
 
+func getDiskKindByStorageMediaKind(
+	diskKind core_protos.EStorageMediaKind,
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/client.go:81]: mediaKind

File: cloud/disk_manager/internal/pkg/clients/nbs/client.go [hunk 2]
```diff
@@ -77,6 +77,37 @@ func getStorageMediaKind(
 	}
 }
 
+func getDiskKindByStorageMediaKind(
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/client.go:80]: getDiskKind

File: cloud/disk_manager/internal/pkg/clients/nbs/interface.go [hunk 3]
```diff
@@ -59,13 +59,14 @@ type DiskModel struct {
...
-	BlockSize      uint32
-	BlocksCount    uint64
-	EncryptionDesc *types.EncryptionDesc
-	CloudID        string
-	FolderID       string
-	BaseDiskID     string
-	IsFillFinished bool
+	BlockSize        uint32
+	BlocksCount      uint64
+	StorageMediaKind types.DiskKind
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/interface.go:64]: StorageMediaKind -> Kind

File: cloud/disk_manager/internal/pkg/clients/nbs/client.go [hunk 4]
```diff
@@ -77,6 +77,37 @@ func getStorageMediaKind(
...
+	diskKind core_protos.EStorageMediaKind,
+) (types.DiskKind, error) {
+
+	switch diskKind {
+	case core_protos.EStorageMediaKind_STORAGE_MEDIA_DEFAULT:
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/client.go:85]: Lets return error here

File: cloud/disk_manager/internal/pkg/services/disks/migrate_disk_task.go [hunk 5]
```diff
@@ -333,17 +333,6 @@ func (t *migrateDiskTask) scheduleReplicateTask(
 	execCtx tasks.ExecutionContext,
 ) error {
 
-	diskMeta, err := t.resourceStorage.GetDiskMeta(ctx, t.request.Disk.DiskId)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/services/disks/migrate_disk_task.go:336]: Do we still need resourceStorage here?

Comment (BarkovBG) [reply cloud/disk_manager/internal/pkg/services/disks/migrate_disk_task.go:336]: yup

https://github.com/ydb-platform/nbs/blob/68ce0e6c84cf3be0a1ac57db5d84491e8a96ee43/cloud/disk_manager/internal/pkg/services/disks/migrate_disk_task.go#L411

https://github.com/ydb-platform/nbs/blob/68ce0e6c84cf3be0a1ac57db5d84491e8a96ee43/cloud/disk_manager/internal/pkg/services/disks/migrate_disk_task.go#L499

**Review by leftmain — APPROVED** [review]

**Review by SvartMetal — APPROVED** [review]

**Pull request merged**